### PR TITLE
NMS-8651: Modified uei.opennms.org/nodes/pathOutage event definition

### PR DESCRIPTION
**Pull request by StCyr**
Modified uei.opennms.org/nodes/pathOutage event definition to reduce on critical path rather than nodes with unknown status behind it.

JIRA: http://issues.opennms.org/browse/NMS-8651
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS988

### CODE REVIEW

**Comment by soleger** [comment]
@jeffgdotorg @agalue @pbrane Does reducing based on the critical path node and service make sense here?

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml [hunk 1]
```diff
@@ -909,18 +909,16 @@
     <uei>uei.opennms.org/nodes/pathOutage</uei>
     <event-label>OpenNMS-defined node event: pathOutage</event-label>
     <descr>
-                        &lt;p&gt;The state of node %parm[nodelabel]% is unknown
-                        because the critical path to the node is down.&lt;/p&gt;
-                        &lt;p&gt;This event is generated when node outage processing
-                        determines that the critical path IP address/service for
-                        this node is not responding..&lt;/p&gt;
+                        &lt;p&gt;A path outage is ongoing. The status of one or several
```
Comment (soleger) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: Can you undo these wording changes? I think the original wording is better. You can append the sentence with the link to the Path Outages list but leave the rest of the text as it was.

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml [hunk 2]
```diff
@@ -909,18 +909,16 @@
...
+                        node(s) might therefore be unknown.&lt;/p&gt;
+                        &lt;p&gt;More information on the affected node(s) can be found at
+                        the &lt;a href="opennms/pathOutage/index.jsp"&gt;Path Outages&lt;/a&gt; page.&lt;/p&gt;
                 </descr>
     <logmsg dest="logndisplay">
-                        %parm[nodelabel]% path outage. Critical path =
-                        %parm[criticalPathIp]% %parm[criticalPathServiceName]%
+                        path outage. Critical path = %parm[criticalPathIp]% %parm[criticalPathServiceName]%
```
Comment (soleger) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: Can you revert this change? It appears that your only change was to get rid of the ```%parm[nodelabel]%``` at the beginning which I assume is a typo.

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml (hunk 1 above)
Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: done

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml (hunk 2 above)
Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: done

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml [hunk 3]
```diff
@@ -914,13 +914,15 @@
...
+                        &lt;p&gt;More information on the affected node(s) can be found at
+                        the &lt;a href="opennms/pathOutage/index.jsp"&gt;Path Outages&lt;/a&gt; page.&lt;/p&gt;
                 </descr>
     <logmsg dest="logndisplay">
                         %parm[nodelabel]% path outage. Critical path =
                         %parm[criticalPathIp]% %parm[criticalPathServiceName]%
                 </logmsg>
     <severity>Major</severity>
-    <alarm-data reduction-key="%uei%:%dpname%:%nodeid%" alarm-type="3" auto-clean="false"/>
+    <alarm-data reduction-key="%uei%:%parm[criticalPathIp]%:%parm[criticalPathServiceName]%" alarm-type="3" auto-clean="false"/>
```
Comment (soleger) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: Can you please add %dpname% back to the reduction key? This field may come into play now that we support distributed monitoring with the Minion framework.

Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: No, sorry. That's the main point of this PR: To get only one path outage alarm per path outage, and not one alarm per node affected by a path outage.

Cyrille

Comment (soleger) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: The %dpname% token represents the "distributed poller" responsible for initiating the event. It is not part of the node, it represents the OpenNMS system responsible for generating the event so it still needs to be included in the reduction key.

This terminology is evolving as we finally implement our distributed monitoring concept (after 15 long years of incubation) with the work on Minion and on HA clustering for OpenNMS. We won't use the "distributed poller" term any more but if you see "monitoring system" in the UI or database, that refers to the same concept of the management system that detected the event.

Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.events.xml]: Ok, but in my system, ONMS 18.0.1 not using distributed pollers, I think %dpname% translate to the display name of the node affected by the path outage...

....

Ok, I may be wrong indeed....

I'll add %dpname% back

**Comment by soleger** [comment]
These changes were rebased and merged. Closing this PR.
